include/dlaf_c/grid.h

```diff
@@ -0,0 +1,67 @@
+//
+// Distributed Linear Algebra with Future (DLAF)
+//
+// Copyright (c) 2018-2023, ETH Zurich
+// All rights reserved.
+//
+// Please, refer to the LICENSE file in the root directory.
+// SPDX-License-Identifier: BSD-3-Clause
+//
+
+#pragma once
+
+#include <mpi.h>
+
+#include "dlaf_c/utils.h"
+
+/// Create communication grid
+///
+/// Grids created here are indexed starting from INT_MAX, to avoid clashes with
+/// BLACS contexts (which start from 0)
+///
+/// Grid ordering can be column-major ("C") or rowp-major ("R"). Row-major is the
+/// default ordering.
+///
+/// @param MPI communicator
+/// @param nprow Number of process rows in the communicator grid
+/// @param npcol Number of process columns in the communicator grid
+/// @param order Grid ordering
+/// @return DLA-Future context
+DLAF_EXTERN_C int dlaf_create_grid(MPI_Comm comm, int nprow, int npcol, char order);
+
+/// Free communicator grid for the given context
+///
+/// This function only frees the DLA-Future grids stored internally. If you
+/// created a BLACS grid with blacs_gridinit you still need to call blacs_gridexit
+///
+/// @param context BLACS or DLA-Future context associated to the grid being released
+DLAF_EXTERN_C void dlaf_free_grid(int context);
+
+#ifdef DLAF_WITH_SCALAPACK
+
+/// Create communication grid from BLACS context
+///
+/// Grids created here are indexed by their corresponding BLACS context
+/// @param blacs_ctxt.
+///
+/// The grid ordering is automatically inferred from the BLACS grid ordering.
+/// Only row-major and column-major grids are supported (created with
+/// blacs_gridinit). Grids created with blacs_gridmap are not supported.
+///
+/// @param blacs_ctxt BLACS context
```

Comment:
I'm not sure we've had cases like this before, but do you think it would be useful to mention in the docstring that this function is only available if `DLAF_WITH_SCALAPACK=ON`? Also, docs should be built with `DLAF_WITH_SCALAPACK` defined for this to show up.

Reply:
> Also, docs should be built with DLAF_WITH_SCALAPACK defined for this to show up.

Good point, this is done now with `PREDEFINED += DLAF_WITH_SCALAPACK`.

Reply:
> but do you think it would be useful to mention in the docstring that this function is only available if DLAF_WITH_SCALAPACK=ON

Yes, I think it would be useful! I added this under a `@remark`.